test/bindings/dom.js

```diff
@@ -80,3 +83,96 @@ describe('getAttributes', function() {
     assert.deepEqual(ret.args, l10nArgs);
   });
 }); 
+
+describe.skip('translateFragment', function() {
+  it('should translate fragment', function() {
+    let dom = new L20n.LocalizationObserver();
+    const div = document.createElement('div');
+    div.innerHTML = `
+      <p data-l10n-id="key1"></p>
+      <p data-l10n-id="key2"></p>
+      <div>
+        <p data-l10n-id="key3"></p>
+      </div>
+    `;
+
+    function requestBundles() {
+      let rb = new L20n.ResourceBundle('en-US', [
+        './locales/test.ftl'
+      ], {
+        './locales/test.ftl': `
+key1 = Value 1
+key2 = Value 2
+key3 = Value 3
+      `
+      });
+      return Promise.resolve([
+        rb
+      ]);
+    }
+
+    function createContext(lang) {
+      return new L20n.MessageContext();
+    }
+    const loc = new L20n.HTMLLocalization(requestBundles, createContext);
+    dom.set('main', loc);
+    return dom.translateFragment(div).then(() => {
+      dom.localizations.delete('main');
```

Comment:
I think it good to either clean up _or_ recreate instances for each test, but doing both is probably overkill and even a bit confusing :)